File: leetcode/49.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups: Dict[str, List[str]] = {}

        for s in strs:
            sorted_s = "".join(sorted(s))

            if sorted_s in groups:
                groups[sorted_s] = groups[sorted_s] + [s]
            else:
                groups[sorted_s] = [s]

        return list(groups.values())

File: leetcode/49.kt
class Solution {
    fun groupAnagrams(strs: Array<String>): List<List<String>> {
        val map: MutableMap<String, List<String>> = mutableMapOf()

        for (str in strs) {
            val key: String = str.toList().sorted().joinToString("")
            map[key] = map.getOrDefault(key, listOf()) + str
        }

        val out: MutableList<List<String>> = mutableListOf()
        for (key in map.keys) {
            out.add(map[key]!!)
        }
        return out
    }
}
